types.go

```diff
@@ -51,13 +51,10 @@ func (t TypeID) Equal(other Type) bool {
 
 type AnyType struct{}
 
-func NewAnyType() AnyType {
-	return AnyType{}
-}
+var TheAnyType = AnyType{}
```

Comment:
Probably want to box as `Type` here and for the other declarations:
```suggestion
var TheAnyType Type = AnyType{}
```

Reply:
That would be problematic for the constructors: they return the concrete type
e.g:
```cadence
func NewAnyType() AnyType
```